File: examples/python/declarations.py
"""Module-level bindings used to exercise the Python declaration listing.

Python has no constant: every binding here is reported as a variable, because
inferring a constant from an upper-case name would be a guess dressed as a fact.
"""

# MAX_RETRIES caps the retry loop.
MAX_RETRIES = 3

# TIMEOUT carries an annotation, which the listed line preserves.
TIMEOUT: float = 1.5

# TABLE is bound to a multi-line literal.
TABLE = {
    "one": 1,
    "two": 2,
}


def described():
    """Return a description, so the file also holds one callable."""
    return "declarations"

File: examples/javascript/declarations.js
// Bodyless top-level bindings used to exercise the JavaScript listing.

// MAX caps the retry loop.
export const MAX = 3;

// counter is mutable, so it binds as a variable rather than a constant.
let counter = 0;

// legacy uses the older binding form.
var legacy = "declarations";

// TABLE is bound to a multi-line literal.
const TABLE = {
  one: 1,
  two: 2,
};

// sum adds two numbers, so the file also holds one callable.
export function sum(a, b) {
  return a + b;
}
